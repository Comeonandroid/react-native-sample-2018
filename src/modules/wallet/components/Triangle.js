import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Triangle from 'react-native-triangle' // eslint-disable-line
import styled, { withTheme } from 'styled-components'

import { View } from '../../../elements'

class TriangleComponent extends PureComponent {
  render() {
    const { direction, width, height, color, children, ...rest } = this.props
    return (
      <Wrapper>
        <Element {...rest}
          width={width}
          height={height}
          color={color}
          direction={direction}
        />
        { children && children }
      </Wrapper>
    )
  }
}

const Wrapper = styled(View) `
  position: relative;
`

const Element = styled(Triangle) `

`

TriangleComponent.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.object,
  color: PropTypes.string,
  direction: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number
}

TriangleComponent.defaultProps = {
  direction: 'down-left',
  width: 22,
  height: 22
}

export default withTheme(TriangleComponent)
