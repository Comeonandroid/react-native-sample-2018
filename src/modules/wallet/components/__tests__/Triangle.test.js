import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import Triangle from '../Triangle'

describe('<Triangle />', () => {
  itBehavesLike('aThemedComponent', { subject: <Triangle /> })
})
