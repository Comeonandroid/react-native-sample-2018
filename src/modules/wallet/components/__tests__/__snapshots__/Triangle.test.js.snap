// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Triangle /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderStyle": "solid",
          "height": 0,
          "width": 0,
        },
        Object {
          "borderBottomColor": "transparent",
          "borderBottomWidth": 0,
          "borderLeftColor": "white",
          "borderLeftWidth": 22,
          "borderRightColor": "transparent",
          "borderRightWidth": 0,
          "borderTopColor": "transparent",
          "borderTopWidth": 22,
        },
        Array [
          Object {},
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`<Triangle /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderStyle": "solid",
          "height": 0,
          "width": 0,
        },
        Object {
          "borderBottomColor": "transparent",
          "borderBottomWidth": 0,
          "borderLeftColor": "white",
          "borderLeftWidth": 22,
          "borderRightColor": "transparent",
          "borderRightWidth": 0,
          "borderTopColor": "transparent",
          "borderTopWidth": 22,
        },
        Array [
          Object {},
          undefined,
        ],
      ]
    }
  />
</View>
`;
